WcaOnRails/lib/results_validators/persons_validator.rb

```diff
@@ -111,6 +114,25 @@ def validate(competition_ids: [], model: Result, results: nil)
           if without_wca_id.select { |p2| p2.name == p.name }.length > 1 && !duplicate_newcomer_names.include?(p.name)
             duplicate_newcomer_names << p.name
           end
+          # Look for obvious person name issues
+          splitted_name = p.name.split
+          if splitted_name.any? { |n| n.downcase == n }
+            @warnings << ValidationWarning.new(:persons, competition_id,
+                                               LOWERCASE_NAME_WARNING,
+                                               name: p.name)
+          end
+          if splitted_name.length > 2
+            if splitted_name[0, splitted_name.length-2].any? { |n| n.length == 1 }
+              @warnings << ValidationWarning.new(:persons, competition_id,
+                                                 MISSING_ABBREVIATION_PERIOD_WARNING,
+                                                 name: p.name)
+            end
+          end
+          if p.name[0, 2].include?(" ") || p.name[0, 2].include?(".") || p.name.reverse[0, 2].include?(" ") || p.name.reverse[0, 2].include?(".")
```

Comment:
This seems convoluted. Couldn't you just take the split name and check if any of them are length 1?

Reply:
This should only apply to first and last names, since abbreviations are allowed for middle names. It should also catch abbreviations with a period, which have length 2.